packages/mini-browser/src/browser/location-mapper-service.ts

```diff
@@ -109,7 +109,7 @@ export class HttpsLocationMapper implements LocationMapper {
  * Location mapper for locations without a scheme.
  */
 @injectable()
-export class LocationWithoutSchemeMapper implements LocationMapper {
+export class RemoteLocationWithoutSchemeMapper implements LocationMapper {
```

Comment:
I'd propose leaving the class names as they are, and prefix the electron implementation with `electron` in the module name, and `Electron` prefix for the class name. 

Something like this. Would this work for you?

![Screen Shot 2020-02-04 at 08 25 46](https://user-images.githubusercontent.com/1405703/73722936-023c6d80-4728-11ea-80ea-639e1444b7b8.jpg)

Reply:
Done, is it better?

Reply:
>  is it better?

Reply:
In case I missed something ;)

Reply:
Looks great, do we need the empty electron backend module? Just asking though.

Reply:
I don't know how to tell the generators to not include a module based on the target. Simpliest workaround is an empty module. Unless there is a way?

Reply:
> I don't know how to tell the generators to not include a module based on the target

Sorry, I do not understand it. What generators do you mean?

Reply:
I mean the generators involved when we do `theia build --mode development`, mounting all the module containers based on the target (electron/browser) from what we define in the `theiaExtensions` field of our packages.

Reply:
Got it. Why do not you omit the `backendElectron` entry? It is not mandatory. See here for instance, although this is the other way around:
https://github.com/eclipse-theia/theia/blob/cc0da4e69f8141122496c119bb72f4df46584e76/packages/core/package.json#L60-L64

Reply:
Issue is that if I define `backend` without defining `backendElectron` then the generators will default to include the module defined by `backend` when target is electron. I mean, I could change the generators to accept `null` to avoid defining empty containers...

Reply:
@kittaakos are we fine using empty containers for now?